lib/pool.py

```diff
@@ -185,3 +185,24 @@ def closeall(self):
             self._closeall()
         finally:
             self._lock.release()
+
+class MultiProcessThreadedConnectionPool(ThreadedConnectionPool):
```

Comment:
Nobody would use a specific subclass if they haven't already gone through some painful debug process as you say. So maybe this should be a change to the base class, not the addition of a new class.